runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java

```diff
@@ -121,202 +228,84 @@ public Dimension render(Graphics2D g)
+		if (right != null)
 		{
-			final BufferedImage healthImage = skillIconManager.getSkillImage(Skill.HITPOINTS, true);
-			final int counterHealth = client.getBoostedSkillLevel(Skill.HITPOINTS);
-			final int counterPrayer = client.getBoostedSkillLevel(Skill.PRAYER);
-			final String counterHealthText = Integer.toString(counterHealth);
-			final String counterPrayerText = Integer.toString(counterPrayer);
-
-			renderIconsAndCounters(g, offsetPrayerX, offsetPrayerY, prayerImage, counterPrayerText, PRAYER_LOCATION_X);
-			renderIconsAndCounters(g, offsetHealthX, offsetHealthY, healthImage, counterHealthText, HEALTH_LOCATION_X);
+			right.renderBar(g, offsetRightBarX, offsetRightBarY, height);
 		}
-
 		return null;
 	}
 
-	private static void renderBar(Graphics2D graphics, int x, int y, int max, int current, int height, Color filled)
-	{
-		graphics.setColor(BACKGROUND);
-		graphics.drawRect(x, y, WIDTH - PADDING, height - PADDING);
-		graphics.fillRect(x, y, WIDTH, height);
-
-		final int filledHeight = getBarHeight(max, current, height);
-		graphics.setColor(filled);
-		graphics.fillRect(x + PADDING,
-			y + PADDING + (height - filledHeight),
-			WIDTH - PADDING * OFFSET,
-			filledHeight - PADDING * OFFSET);
-	}
-
-	private static void renderHealingBar(Graphics2D graphics, int x, int y, int max, int current, int height, int heal, Color color)
-	{
-		if (heal <= 0)
-		{
-			return;
-		}
-
-		final int filledCurrentHeight = getBarHeight(max, current, height);
-		int filledHeight = getBarHeight(max, heal, height);
-		graphics.setColor(color);
-
-		if (filledHeight + filledCurrentHeight > height)
-		{
-			final int overHeal = filledHeight + filledCurrentHeight - height;
-			filledHeight = filledHeight - overHeal + OVERHEAL_OFFSET;
-			graphics.setColor(OVERHEAL_COLOR);
-			graphics.fillRect(x + PADDING,
-				y - filledCurrentHeight + (height - filledHeight) + HEAL_OFFSET,
-				WIDTH - PADDING * OVERHEAL_OFFSET,
-				filledHeight - PADDING * OVERHEAL_OFFSET);
-		}
-		else
-		{
-			graphics.fillRect(x + PADDING,
-				y - OVERHEAL_OFFSET - filledCurrentHeight + (height - filledHeight) + HEAL_OFFSET,
-				WIDTH - PADDING * OVERHEAL_OFFSET,
-				filledHeight + OVERHEAL_OFFSET - PADDING * OVERHEAL_OFFSET);
-		}
-	}
-
-	private static int getBarHeight(int base, int current, int size)
+	private int getRestoreValue(String skill)
 	{
-		final double ratio = (double) current / base;
+		final MenuEntry[] menu = client.getMenuEntries();
+		final int menuSize = menu.length;
+		final MenuEntry entry = menuSize > 0 ? menu[menuSize - 1] : null;
+		int restoreValue = 0;
 
-		if (ratio >= 1)
+		if (entry != null && entry.getParam1() == WidgetInfo.INVENTORY.getId())
 		{
-			return size;
-		}
+			final Effect change = itemStatService.getItemStatChanges(entry.getIdentifier());
 
-		return (int) Math.round(ratio * size);
-	}
+			if (change != null)
+			{
+				final StatsChanges statsChanges = change.calculate(client);
```

Comment:
Since this is not reused, move the call into the loop below to remove the unneeded variable definition:

```java
for (final StatChange c : change.calculate(client).getStatChanges())
```